library/src/scene_impl.cxx

```diff
@@ -329,6 +331,102 @@ scene& scene_impl::clear()
   return *this;
 }
 
+//----------------------------------------------------------------------------
+int scene_impl::addLight(const light_state_t& lightState)
+{
+  vtkNew<vtkLight> newLight;
+  newLight->SetLightType(static_cast<int>(lightState.type));
+  newLight->SetPosition(lightState.position.data());
+  newLight->SetColor(lightState.color.data());
+  newLight->SetPositional(lightState.positionalLight);
+  newLight->SetFocalPoint(lightState.position[0] + lightState.direction[0],
+    lightState.position[1] + lightState.direction[1],
+    lightState.position[2] + lightState.direction[2]);
+  newLight->SetIntensity(lightState.intensity);
+  newLight->SetSwitch(lightState.switchState);
+  this->Internals->Window.GetRenderer()->AddLight(newLight);
+  return this->getLightCount() - 1;
+}
+
+//----------------------------------------------------------------------------
+int scene_impl::getLightCount() const
+{
+  vtkLightCollection* lc = this->Internals->Window.GetRenderer()->GetLights();
+  return lc->GetNumberOfItems();
+}
+
+//----------------------------------------------------------------------------
+light_state_t scene_impl::getLight(int index)
+{
+  vtkLightCollection* lc = this->Internals->Window.GetRenderer()->GetLights();
+  vtkLight* light = vtkLight::SafeDownCast(lc->GetItemAsObject(index));
+  if (!light)
+  {
+    throw scene::load_failure_exception("No light at index " + std::to_string(index) + " to get");
```

Comment:
you need to declare a new exception eg light_exception for this. and document the behavior in scene.h